app/client/src/UITelemetry/generateTraces.ts

```diff
@@ -91,3 +91,15 @@ export function wrapFnWithParentTraceContext(parentSpan: Span, fn: () => any) {
   const parentContext = trace.setSpan(context.active(), parentSpan);
   return context.with(parentContext, fn);
 }
+
+export function generateRootSpan(
+  spanName: string,
+  difference: number,
+  spanAttributes: SpanAttributes = {},
+) {
+  const startTime = Date.now();
+  const endTime = startTime + Math.floor(difference);
+
+  const span = startRootSpan(spanName, spanAttributes, startTime);
+  span.end(endTime);
+}
```

Comment:
nit: Seems like a custom implementation for this use case. I would suggest creating a genetic fn that accepts the startTime and duration.

```suggestion
export function startAndEndSpan(
  spanName: string,
  startTime: number
  duration: number,
  spanAttributes: SpanAttributes = {},
) {
  const endTime = startTime + duration;
  const span = startRootSpan(spanName, spanAttributes, startTime);
  span.end(endTime);
}
```